apps/root/src/services/earnService.ts

```diff
@@ -324,6 +337,24 @@ export class EarnService extends EventsManager<EarnServiceData> {
 
     const userStrategy = await this.sdkService.getUserStrategy(strategyId);
 
-    this.updateUserStrategy(userStrategy);
+    if (updateStrategies) {
+      this.updateUserStrategy(userStrategy);
+    }
+
+    return userStrategy;
+  }
+
+  async fetchMultipleEarnPositionsFromStrategy(strategyId: StrategyId) {
+    const userStrategies = this.userStrategies;
+
+    const positionsToFetch = userStrategies.filter((strat) => strat.strategy === strategyId);
+
+    const promises = positionsToFetch.map((position) => this.fetchUserStrategy(position.id, false));
+
+    const results = compact(await Promise.all(promises));
```

Comment:
We should catch any error, so successful promises can continue

Reply:
Oh good catch